misc/json-to-record-converter/src/main/java/io/ballerina/jsonmapper/JsonToRecordMapper.java

```diff
@@ -258,18 +260,44 @@ private static void updateRecordFields(JsonObject jsonObject, boolean isRecordTy
                     .get(entry.getKey()), jsonNodes.get(jsonEscapedFieldToFields.get(entry.getKey())));
             if (!entry.getValue().getValue0().typeName().toSourceCode()
                     .equals(entry.getValue().getValue1().typeName().toSourceCode()) && !isRecordTypeDesc) {
-                List<TypeDescriptorNode> typeDescNodes =
-                        List.of((TypeDescriptorNode) entry.getValue().getValue0().typeName(),
-                                (TypeDescriptorNode) entry.getValue().getValue1().typeName());
+                TypeDescriptorNode node1 = (TypeDescriptorNode) entry.getValue().getValue0().typeName();
+                TypeDescriptorNode node2 = (TypeDescriptorNode) entry.getValue().getValue1().typeName();
+                Token questionMarkToken = AbstractNodeFactory.createToken(SyntaxKind.QUESTION_MARK_TOKEN);
```

Comment:
Shall we move this `questionMarkToken` assignment into the else block (L279) since its used only in that scope?